src/App.tsx

```diff
@@ -1,26 +1,93 @@
-import React from "react";
-import logo from "./logo.svg";
+import { useState } from "react";
 import "./App.css";
+import { InputTask } from "./components/listItem";
+import { MyButton } from "./components/myButton";
+import { MySearching } from "./components/MySearching";
 
 function App() {
+  const [tasks, setTasks] = useState<{ isDone: boolean; title: string }[]>([]);
+  const [inputValue, setInputValue] = useState<string>("");
+  const [searchValue, setSearchValue] = useState<string>("");
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(e.target.value);
+  };
+
+  const handleCreateTask = () => {
+    if (inputValue.trim() === "") return;
+    const newTask = { title: inputValue, isDone: false };
+    setTasks((prev) => [...prev, newTask]);
+    setInputValue("");
+  };
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter") {
+      handleCreateTask();
+    }
+  };
+  const handleDoneClick = (index: number) => {
+    setTasks((prevTasks) =>
+      prevTasks.map((task, i) =>
+        i === index ? { ...task, isDone: true } : task,
+      ),
+    );
+  };
+  const handleDelClick = (index: number)=>{
+    setTasks((prevTasks) => prevTasks.filter((_, i) => i !== index)
+    );
+  };
+  const handleSearching = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchValue(e.target.value);
+  };
+  const filteredTasks = tasks.filter((task) =>
+    task.title.toLowerCase().includes(searchValue.toLowerCase())
+  );
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p className="bg-orange-500">
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <>
+      <div className="bg-[#2B333B] h-[100vh] flex flex-col items-center text-[#E0E0E0]">
```

Comment:
I would create some dictionary (AKA object, AKA map) to handle all the colors used in the app